client.go

```diff
@@ -145,6 +145,9 @@ func NewClientWithOptions(serverURL string, authToken string, options *Options)
 		}
 		ilog.Infof("Using URL '%s'%s", serverURL, tokenStr)
 	}
+	if options.ApplicationName() == "" {
+		ilog.Error("Application name is not set")
```

Comment:
it should be a warning, and written at most once per application run

Reply:
> As the application name is an option per client instance, the `NewClient...` function is called once per client instance lifecycle.

ok

> Error level is chosen because it is the default log level. I agree a warning would be better, but warning will not be shown by default.
Personally, I think that `NewClient` should be also deprecated (or maybe reorganized), now it will always cause an error to be logged. Please add other approvers and let them decide.